### PR TITLE
Backend Routes Link To Database

### PR DESCRIPTION
**Pull request by rparker2003**
## Description / Changes Made

- _Created a Post Validation function to verify the data of a post_
- _Modified get_post, create_post, update_post, and delete_post routes to directly interface with the Firebase Firestore database_
- _Modified get_user_posts to reflect route naming scheme, and to return proper error codes_
- _Reflected files changed in the .gitignore file for proper secrecy of database keys_
- _Added more status codes to reflect proper error codes for the database connection_
- _Included an unpack script written in python to unpack the .pkl.npz file created by model.py_

## Screenshots

image

## How to Test

1. _Run the backend server with python._
2. _Using Postman, add the address http://localhost:5000._
3. _Depending on the action, give the proper route. For ex: /api/posts/<post_id> with method GET. The post_id can be found in the Firebase console in the "posts" collection. An example can be seen in the image above where it navigates to http://localhost:5000/api/posts/m4WdEpyrC32sYs9mRg5t to complete a GET query on that post_id._
4. _Click send and postman will return the query in JSON format._

## Checklist

- [x] I have verified proper error messages are returned on error
- [x] I have verified the post-validation function expects the current database requirements
- [x] I have verified the modified functions can directly access the Firestore database
- [x] I have checked for any potential conflicts with other branches and fixed any merge conflicts.

### CODE REVIEW

**Review by KeenParchment — APPROVED** [review]
I have thoroughly checked the provided checklist and confirm that:

- Proper error messages are returned on encountering errors.
- The post-validation function aligns with current database requirements.
- The modified functions seamlessly interact with the Firestore database.
- Any potential conflicts with other branches have been resolved.

These changes appear to be comprehensive and well-implemented. The addition of the post-validation function and direct database interaction enhances the reliability and efficiency of the system. Moreover, the addition of documentation is appreciated with the comments. 

Thank you for your contribution!

File: .gitignore
Comment (Dhooly) [inline .gitignore:1]: Slightly uncomfortable with putting the exact route to the private key in the .gitignore, but shouldn't pose a security risk unless attacker has internal access to files which is already worst case scenerio

File: backend/__pycache__/status.cpython-310.pyc
Comment (Dhooly) [inline backend/__pycache__/status.cpython-310.pyc:1]: No need to push __pycache__ files. Typically anything thats generated when building or running commands does not need to be pushed

**Review by Dhooly — APPROVED** [review]
### Additions
- The code is well-organized into different functions and routes, enhancing readability and maintainability.
- Good use of comments to explain the purpose and functionality of each function and route.
- The idea of token verification on each request is crucial for ensuring security and data integrity.

### Suggested Improvements
- Remove __pycache__ binary files. Any cache files or generated files should not be pushed unless it's crucial or required to run and build the program/script.
- A bit of repetition in error handling, could use some refactoring
- Post validation could be a little more rigid

**Pull request merged**